### DISCUSSION TITLE
Pydandic don't work like i except

### DISCUSSION THREAD

**Discussion by AlexTraveylan** [fastapi/sqlmodel#937] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import PositiveInt
from sqlalchemy import create_engine
from sqlmodel import Field, Session, SQLModel

from app.adapter.exception.app_exception import AppException


class Personage(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str = Field(max_length=20, min_length=2)
    age: PositiveInt
```

### Description

![GNU81w7WQAM49Eu](https://github.com/tiangolo/sqlmodel/assets/108667536/92967af4-674b-4f98-94eb-8d48ef6d0de1)

Pydantic allow me to do everything i want with data, it is not typesafe at all.

full code here : https://github.com/AlexTraveylan/sql_model_training

### Operating System

Windows

### SQLModel Version

0.0.18

### Python Version

Python 3.12.2

### Additional Context

I love using SQLAlchemy and Pydantic together with FastAPI, I was about to create something similar to SQLModel, however, it does not meet my needs. The models appear to inherit poorly from BaseModel or maybe I am doing it wrong. Anyway, I don't know if it's a bug or not, but I can't use this library because of it.

I would like to contribute rather than create my own thing if possible.

**Comment by joachimhuet** [comment, marked as answer]
When `table=True` the validation is turned off.
Check that thread for alternatives solutions: #52

**Reply by AlexTraveylan** [reply]
Thanks you, i will try again. Maybe it should be writen on the doc ... it's an important feature totaly hidden
